### ISSUE TITLE
protect_from_forgery 

### ISSUE THREAD

**Issue by samaswin** [apokalipto/devise_saml_authenticatable#91]
Please help me to resolve the bug
protect_from_forgery in application controller block the certificate.  How we can skip for this

**Comment by adamstegman** [comment]
It shouldn't; this gem skips that filter: https://github.com/apokalipto/devise_saml_authenticatable/blob/master/app/controllers/devise/saml_sessions_controller.rb#L6

What route is the SAML response coming to? Is that route pointing to the `Devise::SamlSessionsController`?

**Comment by ph0rque** [comment]
I had the same issue, and I fixed it by changing the line in https://github.com/apokalipto/devise_saml_authenticatable/blob/master/app/controllers/devise/saml_sessions_controller.rb#L6

from `skip_before_filter :verify_authenticity_token, raise: false`

to: `skip_before_action :verify_authenticity_token`

I'm not sure why the `raise: false` part is causing the error?

**Comment by adamstegman** [comment]
What is your Rails version?

**Comment by ph0rque** [comment]
4.2.7.1

**Comment by adamstegman** [comment]
We definitely test against Rails 4. By default it's configured with `protect_from_forgery with: :exception`, I'll try testing against other strategies.

**Comment by ph0rque** [comment]
@adamstegman That's what we have in our `application_controller`.

**Comment by adamstegman** [comment]
`raise: false` definitely does not cause problems in Rails 4. I also noticed you changed `before_filter` to `before_action`. These should be the same until Rails 5.1, so I wouldn't anticipate this causing an issue until that is released. What else could this be?

**Comment by timminkov** [comment]
FYI - I had this issue a few weeks ago. I forked and removed the `raise: false` which also fixed the issue for me.

**Comment by adamstegman** [comment]
Can anyone provide an example or test case that reproduces this?

**Comment by awendt** [comment]
@adamstegman I ran into the same issue, in the process of upgrading Rails 4.2.7.1 to 4.2.8 and devise_saml_authenticatable from 1.3.0 to 1.3.1.

When you say "We definitely test against Rails 4", are you aware that Rails disables this type of forgery protection?

**Comment by awendt** [comment]
For the record, I reverted back to 1.3.0 and it works again

**Comment by adamstegman** [comment]
> When you say "We definitely test against Rails 4", are you aware that Rails disables this type of forgery protection in tests?

I wasn't, good catch! 😉 

So I enabled production mode in the apps used under test (they're run in separate processes): https://github.com/apokalipto/devise_saml_authenticatable/compare/rails4—this builds fine. Additionally I upgraded my Rails 4.2 application to v1.3.1 of this gem (from 1.3.0) with no difficulty.

If anyone has some way to reproduce this please show me!

**Comment by awendt** [comment]
@adamstegman I've created a Rails application template here: https://gist.github.com/awendt/698769e64c15fa778311122e940f2df3

Use that to create a minimal Rails app to reproduce:

1. Create 3 files with the corresponding URLs as content:
  - issuer.url
  - slo_target.url
  - sso_target.url
2. Create a `cert.pem` with your IdP's certificate
3. Download the template into the same directory as all above files and run this command:
  ```
  rails _4.2.8_ new saml_app -m template.rb
  ```
4. Start the Rails app:
  ```
  bundle exec rails s
  ```
5. Go to http://localhost:3000/users/saml/sign_in, sign in
6. Observe

Using 1.3.1, this minimal app raises an `ActionController::InvalidAuthenticityToken` while 1.3.0 gets a step further and runs into issues with `attribute-map.yml` (I didn't create that file in the template).

To test 1.3.0, add a specific version to `template.rb:1` and re-run the `rails new` command (with a different target or `rm -rf` the previous target first).

Hope this helps!

**Comment by adamstegman** [comment]
Thank you @awendt, that does help! I successfully reproduced it and figured out why my tests weren't showing the error. I opened #94 to address this.

**Comment by adamstegman** [comment]
Fixed in v1.3.2
